httplug/exceptions.rst

```diff
@@ -18,6 +18,10 @@ TransferException                  something unexpected happened \-
 ================================== ============================= ===================
 
 .. note::
+    By default clients will always return a PSR-7 response instead of throwing a HttpException. Configure your client
```

Comment:
i would say: Write your application to check the response status or use the plugin to make sure the ``HttpException`` is thrown.

configuring the actual client to throw exceptions should be discouraged, it will mess with the plugin chain and logging http errors, among other things.